src/component/Header/Header.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+interface HeaderProps {
+  todos: Todo[];
+  addTodo: (title: string) => void;
+  query: string;
+  setQuery: (query: string) => void;
+  toggleAll: boolean;
+  setToggleAll: (value: boolean) => void;
+}
+
+export const Header: React.FC<HeaderProps> = ({
+  todos,
+  addTodo,
+  query,
+  setQuery,
+  toggleAll,
+  setToggleAll,
+}) => {
+  const handleSubmit = (event: React.SyntheticEvent) => {
+    event.preventDefault();
+    addTodo(query);
+    setQuery('');
+  };
+
+  const handleToggleAll = () => {
```

Comment:
handleToggleAll function should also toggle the completed status of all the tasks.